### PR TITLE
added filter in  knowledgecafe module

### PR DESCRIPTION
**Pull request by akriti12negi**
#2186 

### Description
Added filter the books that i have read in KnowledgeCafe module

### Checklist:
I have performed a self-review of my own code.

### CODE REVIEW

File: app/Models/KnowledgeCafe/Library/Book.php [hunk 1]
```diff
@@ -20,16 +20,20 @@ public function categories()
...
-            ->where(function ($query) use ($filteredString) {
-                if ($filteredString) {
-                    $query->where('title', 'LIKE', "%$filteredString%")
-                        ->orWhere('author', 'LIKE', "%$filteredString%")
-                        ->orWhere('isbn', 'LIKE', "%$filteredString%");
+            ->where(function ($query) use ($searchString) {
+                if ($searchString) {
+                    $query->where('title', 'LIKE', "%$searchString%")
+                        ->orWhere('author', 'LIKE', "%$searchString%")
+                        ->orWhere('isbn', 'LIKE', "%$searchString%");
```
Comment (gautam071) [inline app/Models/KnowledgeCafe/Library/Book.php:36]: What's the reason of changing the variable name?

**Review by gautam071 — CHANGES_REQUESTED** [review]

File: app/Models/KnowledgeCafe/Library/Book.php (hunk 1 above)
Comment (akriti12negi) [reply app/Models/KnowledgeCafe/Library/Book.php]: fixed it

File: app/Models/KnowledgeCafe/Library/Book.php [hunk 2]
```diff
@@ -20,16 +20,20 @@ public function categories()
...
-            ->where(function ($query) use ($filteredString) {
-                if ($filteredString) {
-                    $query->where('title', 'LIKE', "%$filteredString%")
-                        ->orWhere('author', 'LIKE', "%$filteredString%")
-                        ->orWhere('isbn', 'LIKE', "%$filteredString%");
+            ->where(function ($query) use ($filterString) {
+                if ($filterString) {
+                    $query->where('title', 'LIKE', "%$filterString%")
+                        ->orWhere('author', 'LIKE', "%$filterString%")
+                        ->orWhere('isbn', 'LIKE', "%$filterString%");
```
Comment (gautam071) [inline app/Models/KnowledgeCafe/Library/Book.php:36]: Variable name changes why?

**Review by gautam071 — CHANGES_REQUESTED** [review]
Looks good a question

File: app/Models/KnowledgeCafe/Library/Book.php (hunk 2 above)
Comment (akriti12negi) [reply app/Models/KnowledgeCafe/Library/Book.php:36]: changed earlier, this is the original variable name

**Pull request closed without merging**